src/multicellular_stage/CellBodyPlanInternalCalculations.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System;
+using System.Collections.Generic;
+using Godot;
+
+public static class CellBodyPlanInternalCalculations
+{
+    public static Dictionary<Compound, float> GetTotalSpecificCapacity(IEnumerable<CellTemplate> cells,
+        out float nominalCapacity)
+    {
+        nominalCapacity = 0.0f;
+
+        var capacities = new Dictionary<Compound, float>();
+
+        foreach (var cell in cells)
```

Comment:
I'd add a TODO comment here about checking if this can be sped up by going by cell type and then multiplying that by how many cells of that type there are in the bodyplan.